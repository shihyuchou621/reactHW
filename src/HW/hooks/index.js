import HW1 from './HW1';
import HW2 from './HW2';
import HW3 from './HW3';
import HW4 from './HW4';
import HW5 from './HW5';
import HW6 from './HW6';
import HW7 from './HW7';
import HW8 from './HW8';
import HW9 from './HW9';
import HW10 from './HW10';
import HW11 from './HW11';
import HW12 from './HW12';
import HW13 from './HW13';
import HW14 from './HW14';
import HW15 from './HW15';
import HW16 from './HW16';
import HW17 from './HW17';
import HW18 from './HW18';
import HW19 from './HW19';
import HW20 from './HW20';
import HW21 from './HW21';
import HW22 from './HW22';
import HW23 from './HW23';
import HW25 from './HW25';
import HW26 from './HW26';
import HW27 from './HW27';

export default {
  HW1,
  HW2,
  HW3,
  HW4,
  HW5,
  HW6,
  HW7,
  HW8,
  HW9,
  HW10,
  HW11,
  HW12,
  HW13,
  HW14,
  HW15,
  HW16,
  HW17,
  HW18,
  HW19,
  HW20,
  HW21,
  HW22,
  HW23,
  HW25,
  HW26,
  HW27,
};