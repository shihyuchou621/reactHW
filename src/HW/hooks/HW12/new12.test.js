import React from 'react';
import { render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './new12';

afterEach(cleanup);

test('content 1 should change correctly', () => {
  const { container, getByText } = render(<HW />);
  userEvent.click(getByText("tag1"));
  expect(container).toMatchSnapshot();
});

test('content 2 should change correctly', () => {
  const { container, getByText } = render(<HW />);
  userEvent.click(getByText("tag2"));
  expect(container).toMatchSnapshot();
});

test('content 3 should change correctly', () => {
  const { container, getByText } = render(<HW />);
  userEvent.click(getByText("tag3"));
  expect(container).toMatchSnapshot();
});

test('content 4 should change correctly', () => {
  const { container, getByText } = render(<HW />);
  userEvent.click(getByText("tag4"));
  expect(container).toMatchSnapshot();
});
