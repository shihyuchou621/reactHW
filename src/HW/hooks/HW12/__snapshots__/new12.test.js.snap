// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content 1 should change correctly 1`] = `
<div>
  <div
    class="HW12"
  >
    <h2>
      標籤
    </h2>
    <div
      class="tagName choice"
    >
      tag1
    </div>
    <div
      class="tagName "
    >
      tag2
    </div>
    <div
      class="tagName "
    >
      tag3
    </div>
    <div
      class="tagName "
    >
      tag4
    </div>
    <div
      class="content"
    >
      lorem 1
    </div>
  </div>
</div>
`;

exports[`content 2 should change correctly 1`] = `
<div>
  <div
    class="HW12"
  >
    <h2>
      標籤
    </h2>
    <div
      class="tagName "
    >
      tag1
    </div>
    <div
      class="tagName choice"
    >
      tag2
    </div>
    <div
      class="tagName "
    >
      tag3
    </div>
    <div
      class="tagName "
    >
      tag4
    </div>
    <div
      class="content"
    >
      lorem 2
    </div>
  </div>
</div>
`;

exports[`content 3 should change correctly 1`] = `
<div>
  <div
    class="HW12"
  >
    <h2>
      標籤
    </h2>
    <div
      class="tagName "
    >
      tag1
    </div>
    <div
      class="tagName "
    >
      tag2
    </div>
    <div
      class="tagName choice"
    >
      tag3
    </div>
    <div
      class="tagName "
    >
      tag4
    </div>
    <div
      class="content"
    >
      lorem 3
    </div>
  </div>
</div>
`;

exports[`content 4 should change correctly 1`] = `
<div>
  <div
    class="HW12"
  >
    <h2>
      標籤
    </h2>
    <div
      class="tagName "
    >
      tag1
    </div>
    <div
      class="tagName "
    >
      tag2
    </div>
    <div
      class="tagName "
    >
      tag3
    </div>
    <div
      class="tagName choice"
    >
      tag4
    </div>
    <div
      class="content"
    >
      lorem 4
    </div>
  </div>
</div>
`;
